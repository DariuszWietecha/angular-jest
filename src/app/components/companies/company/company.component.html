<div class="card-body">
  <h4 class="card-title">{{ company.name }}</h4>

  <dt>Name:</dt>
  <dd>{{ company.name }}</dd>

  <dt>Logo:</dt>
  <dd>{{ company.logoUrl }}</dd>

  <dt>E-mail:</dt>
  <dd>{{ company.email }}</dd>

  <dt>Categories:</dt>
  <dd>
    <ul>
      <li *ngFor="let category of company.categories">{{category.name}}</li>
    </ul>
  </dd>

  <a [routerLink]="['/companies/edit', company.id]" class="btn btn-primary"  *ngIf="showEditButton === true">Edit</a>
</div>