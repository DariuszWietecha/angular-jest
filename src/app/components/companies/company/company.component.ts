import { Component, Input, HostBinding } from '@angular/core';
import { ICompany } from '../../../services/companies.service';

@Component({
  selector: 'app-company',
  templateUrl: './company.component.html',
  styleUrls: ['./company.component.scss']
})

export class CompanyComponent {
  @HostBinding('class') className = 'card col-4';
  @Input() company: ICompany;
  @Input() showEditButton = true;
  constructor() { }
}
