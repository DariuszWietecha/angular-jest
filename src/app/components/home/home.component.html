<div class="row">
	<div class="col-3">
		<h4>Categories</h4>
		<ul class="list-group">
			<a [routerLink]="['']" class="list-group-item list-group-item-action"
				[ngClass]="{active: categoryId == null}">All</a>
			<a *ngFor="let category of categories" [routerLink]="['/views', category.id]"
				class="list-group-item list-group-item-action"
				[ngClass]="{active: category.id === categoryId}">{{ category.name }}</a>
		</ul>
	</div>
	<div class="col-9">
		<app-list
			[items]="companies"
			[showCreateButton]="false"
			[title]="'Companies'">
				<ng-template let-item="item">
						<app-company [company]="item" [showEditButton]="false"></app-company>
				</ng-template> 
		</app-list>
	</div>