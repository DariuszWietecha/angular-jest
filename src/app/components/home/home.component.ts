import { ActivatedRoute } from '@angular/router';
import { Component, OnInit } from '@angular/core';
import { ViewsService } from '../../services/views.service';
import { ICompany } from '../../services/companies.service';
import { ICategory } from '../../services/categories.service';

@Component({
  selector: 'app-home',
  templateUrl: './home.component.html',
  styleUrls: ['./home.component.scss']
})
export class HomeComponent implements OnInit {
  categories: ICategory[];
  companies: ICompany[];
  categoryId;

  constructor(
    private route: ActivatedRoute,
    private viewsService: ViewsService) { }

  ngOnInit() {
    this.route.paramMap.subscribe(params => {
      this.categoryId = params.get('categoryId');

      this.viewsService.get(this.categoryId)
        .subscribe((data) => {
          this.companies = data.companies;
          this.categories = data.categories;
          this.categoryId = data.categoryId;
        });
    });
  }
}
