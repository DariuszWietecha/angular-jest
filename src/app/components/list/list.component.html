<div class="text-right" *ngIf="showCreateButton === true">
	<a [routerLink]="['/', createEndpoint, 'create']" class="btn btn-primary" >Create</a>
</div>

<h4>{{ title }}</h4>
<div class="d-flex flex-wrap">
  <div *ngFor="let item of items" class="card col-4">
    <template [ngTemplateOutletContext]='{item: item}' [ngTemplateOutlet]="templateVariable"></template>
  </div>
</div>