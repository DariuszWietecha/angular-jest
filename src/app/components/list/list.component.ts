import { Component, Input, ContentChild, TemplateRef } from '@angular/core';

@Component({
  selector: 'app-list',
  templateUrl: './list.component.html',
  styleUrls: ['./list.component.scss']
})
export class ListComponent {
  @Input() createEndpoint = '';
  @Input() items: any[];
  @Input() showCreateButton = true;
  @Input() title: string;
  @ContentChild(TemplateRef, {static: true}) templateVariable: TemplateRef<any>;

}
