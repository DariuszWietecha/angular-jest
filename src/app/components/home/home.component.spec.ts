import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { HttpClientModule } from '@angular/common/http';
import { RouterTestingModule } from '@angular/router/testing';
import { ViewsService } from '../../services/views.service';

import { CompanyComponent } from '../companies/company/company.component';
import { HomeComponent } from './home.component';
import { ListComponent } from '../list/list.component';

describe('HomeComponent', () => {
  let component: HomeComponent;
  let fixture: ComponentFixture<HomeComponent>;
  const categoriesListMock = require('../../../test/data/categories/categories-get.json');
  const comapaniesListMock = require('../../../test/data/companies/companies-get.json');
  const categoryId = 'a0308830-660b-11ea-9266-adf5decb4537';

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ CompanyComponent, HomeComponent, ListComponent ],
      imports: [HttpClientModule, RouterTestingModule],
      providers: [ViewsService],
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(HomeComponent);
    component = fixture.componentInstance;
    component.categories = categoriesListMock;
    component.companies = comapaniesListMock;
    component.categoryId = categoryId;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('snapShot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
