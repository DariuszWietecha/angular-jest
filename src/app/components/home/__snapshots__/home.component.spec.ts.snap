// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeComponent snapShot 1`] = `
<app-home
  categories={[Function Array]}
  categoryId="null"
  companies={[Function Array]}
  route={[Function ActivatedRoute]}
  viewsService={[Function ViewsService]}
>
  <div
    class="row"
  >
    <div
      class="col-3"
    >
      <h4>
        Categories
      </h4>
      <ul
        class="list-group"
      >
        <a
          class="list-group-item list-group-item-action active"
          href="/"
          ng-reflect-klass="list-group-item list-group-ite"
          ng-reflect-ng-class="[object Object]"
          ng-reflect-router-link=""
        >
          All
        </a>
        
        <a
          class="list-group-item list-group-item-action"
          href="/views/a0308830-660b-11ea-9266-adf5decb4537"
          ng-reflect-klass="list-group-item list-group-ite"
          ng-reflect-ng-class="[object Object]"
          ng-reflect-router-link="/views,a0308830-660b-11ea-9266"
        >
          Big Category
        </a>
        <a
          class="list-group-item list-group-item-action"
          href="/views/eae597d0-660b-11ea-9266-adf5decb4537"
          ng-reflect-klass="list-group-item list-group-ite"
          ng-reflect-ng-class="[object Object]"
          ng-reflect-router-link="/views,eae597d0-660b-11ea-9266"
        >
          Small Category
        </a>
      </ul>
    </div>
    <div
      class="col-9"
    >
      <app-list
        ng-reflect-items="[object Object],[object Object"
        ng-reflect-show-create-button="false"
        ng-reflect-title="Companies"
      >
        
        <h4>
          Companies
        </h4>
        <div
          class="d-flex flex-wrap"
        >
          
          <div
            class="card col-4"
          >
            <template
              ng-reflect-ng-template-outlet="[object Object]"
              ng-reflect-ng-template-outlet-context="[object Object]"
            />
            <app-company
              class="card-body"
              ng-reflect-company="[object Object]"
              ng-reflect-show-edit-button="false"
            >
              <div
                class="card-body"
              >
                <h4
                  class="card-title"
                >
                  Great Company
                </h4>
                <dt>
                  Logo:
                </dt>
                <dd>
                  Great cube
                </dd>
                <dt>
                  E-mail:
                </dt>
                <dd>
                  mail@greatcompany.com
                </dd>
                <dt>
                  Categories:
                </dt>
                <dd>
                  <ul>
                    
                    <li>
                      Big Category
                    </li>
                  </ul>
                </dd>
                
              </div>
            </app-company>
          </div>
          <div
            class="card col-4"
          >
            <template
              ng-reflect-ng-template-outlet="[object Object]"
              ng-reflect-ng-template-outlet-context="[object Object]"
            />
            <app-company
              class="card-body"
              ng-reflect-company="[object Object]"
              ng-reflect-show-edit-button="false"
            >
              <div
                class="card-body"
              >
                <h4
                  class="card-title"
                >
                  Small Company
                </h4>
                <dt>
                  Logo:
                </dt>
                <dd>
                  Small ball
                </dd>
                <dt>
                  E-mail:
                </dt>
                <dd>
                  mail@smallcompany.com
                </dd>
                <dt>
                  Categories:
                </dt>
                <dd>
                  <ul>
                    
                    <li>
                      Quite big company
                    </li>
                  </ul>
                </dd>
                
              </div>
            </app-company>
          </div>
        </div>
      </app-list>
    </div>
  </div>
</app-home>
`;
